File: practice/algorithms/strings/funny_string/funny_string.py
def is_funny(i_str):
    s = len(i_str)
    for CNT in range(s - 1):
        if abs(ord(i_str[CNT + 1]) - ord(i_str[CNT])) != abs(ord(i_str[s - 1 - CNT]) - ord(i_str[s - 2 - CNT])):
            return "Not Funny"
    return "Funny"
t = int(raw_input())
for _ in range(t):
    s = raw_input()
    print is_funny(s)

File: practice/algorithms/strings/funny_string/funny_string.hs
import Data.Char
import Data.Function
import Control.Monad

isFunny :: String -> Bool
isFunny =  and . (\s -> zipWith ((==) `on` abs) s (reverse s)) . (\s -> zipWith ((-)) (tail s) s) . map ord

main = do
    t <- readLn :: IO Int
    forM_ [1..t] $ \_ -> do
        s <- getLine
        if isFunny s
            then putStrLn "Funny"
            else putStrLn "Not Funny"
